scripts/client.js

```diff
@@ -1,47 +1,72 @@
+/* @flow */
+
 import KintoClient from "kinto-client";
 import endpoint from "kinto-client/lib/endpoint";
 import { isHTTPok } from "./utils";
 
 
-let client;
+let client: ?KintoClient;
 
-function getAuthHeader(session) {
-  const {authType, credentials} = session;
+function getAuthHeader(session: Object): ?string {
+  const {authType, credentials}: {
+    authType: string,
+    credentials: Object
+  } = session;
   switch(authType) {
     case "fxa": {
-      const {token} = credentials;
+      const {token}: {token: string} = credentials;
       return "Bearer " + token;
     }
     case "basicauth": {
-      const {username, password} = credentials;
+      const {username, password}: {
+        username: string,
+        password: string,
+      } = credentials;
       return "Basic " + btoa([username, password].join(":"));
     }
   }
 }
 
-export function setupClient(session) {
-  const {server} = session;
+export function setupClient(session: Object): KintoClient {
+  const {server}: {server: string} = session;
```

Comment:
Hmm, on second thought, this might not be possible because of the call to `getAuthHeader` below, which has two very different signatures depending on context..